===================================================================
RCS file: /cvsroot/tar/cvsroot/tar/tar/src/names.c,v
retrieving revision 1.36
retrieving revision 1.37
diff -u -r1.36 -r1.37
--- tar/tar/src/names.c	2003/11/17 07:39:33	1.36
+++ tar/tar/src/names.c	2003/12/25 10:19:40	1.37
@@ -1021,13 +1021,8 @@
 
       for (p = file_name + prefix_len; *p; )
 	{
-	  if (p[0] == '.')
-	    {
-	      if (p[1] == '.' && (ISSLASH (p[2]) || !p[2]))
-		prefix_len = p + 2 - file_name;
-	      else if (ISSLASH (p[1]))
-		prefix_len = p + 1 - file_name;
-	    }
+          if (p[0] == '.' && p[1] == '.' && (ISSLASH (p[2]) || !p[2]))
+	    prefix_len = p + 2 - file_name;
 	  
 	  do
 	    {
@@ -1072,8 +1067,8 @@
 	  };
 	  WARN ((0, 0, _(diagnostic[link_target])));
 	}
-
-      p = ISSLASH (file_name[strlen(file_name)-1]) ? "./" : ".";
+      
+      p = ".";
     }
 
   return (char *) p;
